integration_tests/test_dbapi.py

```diff
@@ -372,3 +373,13 @@ def test_invalid_query_throws_correct_error(trino_connection):
             """,
             params=(3,),
         )
+
+
+def test_eager_loading_cursor_description(trino_connection):
```

Comment:
I would explicitly check that the cursor description is as expected before and after fetching a row.